homework02/fran/string.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+#include <memory>
+#include <iostream>
+#include <string>
+
+class String
+{
+public:
+	String();
+	String(const size_t capacity);
+	String(const char* data);
+	String(const String& other);
+	
+	bool empty() const;
+	
+	size_t size() const;
+	size_t capacity() const;
+	char* c_str() const;
+	
+	String& operator=(String other);
+	String& operator=(const std::string& data);
+	
+	String& operator+(const String& str);
+	
+	friend String operator+(const char* data, const String& other)
+	{
+		return String(data) + other;
+	}
+	
+	bool operator==(const String& other) const;
+	
+	bool operator==(const char* data)
```

Comment:
Do you need this operator? You have a constructor from `const char*` already so I would expect the above overload to be called